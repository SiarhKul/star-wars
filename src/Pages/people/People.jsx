import React from 'react';
import { useSelector } from 'react-redux';

import { ButtonLoadMore } from '../../components';
import { getAbbreviation } from '../../utils';

export const People = () => {
  const people = useSelector(state => state.app.people);

  return (
    <main className="main-people">
      <div className="cards-container">
        {people.map(person => (
          <div className="card" key={person.name}>
            <div className="gab">
              <div className="avatar">
                <div className="avatar__abbr">
                  {getAbbreviation(person.name)}
                </div>
              </div>
              <div className="body">
                <p className="body__text">Name:{person.name}</p>
                <p className="body__text">Birth year:{person.birth_year}</p>
                <p className="body__text">Gender:{person.gender}</p>
              </div>
            </div>
          </div>
        ))}
      </div>
      <ButtonLoadMore />
    </main>
  );
};
