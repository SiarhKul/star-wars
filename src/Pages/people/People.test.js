import React from "react";
import { People } from "./People";
import * as redux from "react-redux";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => ({
			location: {
				pathname: "/people",
			},
		})),
	};
});

describe("<People/>", () => {
	const people = () => shallow(<People />);
	let component;
	let spyOnUseSelectorIsLoadedPeople;
	let spyOnUseSelectorContenCards;

	beforeEach(() => {
		spyOnUseSelectorIsLoadedPeople = jest
			.spyOn(redux, "useSelector")
			.mockReturnValue([{}]);
		spyOnUseSelectorContenCards = jest
			.spyOn(redux, "useSelector")
			.mockReturnValue({ isLoadedPeople: true });
		component = people();
	});

	it("render snapshot <People/> ", () => {
		expect(component).toMatchSnapshot();
	});

	it("should render <People/>", () => {
		expect(component.find("Loader")).toHaveLength(1);
	});

	it("should render alert message in <People/>", () => {
		expect(component.find("ToastContainer")).toHaveLength(1);
	});

	it("should render <People/>", () => {
		expect(component.find("Cards")).toHaveLength(1);
	});

	it("should render ButtonLoadMore in <People/> if  all data has been loaded", () => {
		expect(component.find("ButtonLoadMore").exists()).toBe(false);
	});
});
