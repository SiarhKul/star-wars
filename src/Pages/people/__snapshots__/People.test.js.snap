// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<People/> render snapshot <People/>  1`] = `
<main
  className="main-people"
>
  <Loader />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <Cards
    BodyComponent={[Function]}
    contenCards={
      Object {
        "isLoadedPeople": true,
      }
    }
    name="name"
    path="/people"
  />
</main>
`;
